idb/postgres/postgres_migrations.go

```diff
@@ -31,24 +32,28 @@ type MigrationState struct {
 }
 
 // A migration function should take care of writing back to metastate migration row
-type migrationFunc func(*PostgresIndexerDb, *MigrationState) error
+type postgresMigrationFunc func(*PostgresIndexerDb, *MigrationState) error
 
 type migrationStruct struct {
-	migrate migrationFunc
+	migrate postgresMigrationFunc
 
-	// The system should wait for this migration to finish before trying to import new data or serve data.
-	preventStartup bool
+	// Description of the migration
+	description string
```

Comment:
still need to bring back the preventStartup functionality as prevent-api-access or prevent-db-access

Reply:
We don't need to block access if _any_ migration is running, just some of them. Some of them were extensively designed to run in background for while the system is available.

Reply:
It should be straight forward to add that feature back in, so I'll do that. But I'm not sure if it should be enabled. If we're changing things in the DB, wouldn't we see inconsistent results?

I guess I'm not sure how to judge how "bad" it would be to knowingly return the wrong TxID or AssetConfig object.

You got me thinking about whats actually going on in the background, and I think I may have introduced a bug where the importer will now start importing earlier than before. It definitely needs that "migration-is-blocking" flag.

Reply:
it might be possible to have two attributes of a migration (prevent API, prevent import), but right now it's probably simplest to count them as the same; OR, prevent-api can be a per-migration attribute and any pending or active migration could prevent import. I think preventing import is less of a problem than preventing API access. With the txid migration query-by-txid wouldn't work (but it wouldn't work before the migration either) but other queries would work fine, so disabling the API during that migration would have prevented otherwise normal operation of the indexer. The txid migration would also be safe to run while import is operating. But, unless we carefully add both attributes to each migration, right now I'd vote for having a prevent-API flag and preventing import if there are any running migrations. And the prevent-API flag blocks if any active _or pending_ migration has it set, but after you're past the last one with it set having other migrations doesn't prevent the API from running.

Reply:
How about this... the migration process has 2 attributes about the progress:
* Migration Is Running
* DB Unavailable (new name for preventStartup)

We add a new CLI argument / config value to toggle whether or not queries are allowed during a migration.

Reply:
If a migration knows it is safe to run queries while the migration is running (e.g. the txid fixup), why add an option to override that and not do queries anyway?

Reply:
The DB is technically in an inconsistent state if a migration is running, we're returning data which we know may not be accurate. I don't think we can say for certain that it's safe to return this data since it depends on the application and user. I'm imagining a Tarantino movie with a high stakes transaction that they're checking on and it doesn't get returned because the txid was wrong, that crypto accountant really hopes their indexer admin set the more conservative "block-queries-during-migration: true" setting.

Reply:
Ok, a super-cautious mode makes sense to me now, thanks. I was thinking in terms of restarting `algorand-indexer` with a new version and it starts migrations, it could serve data right away and be no more wrong than it was a minute before the restart, but it could also be super cautious and say, "I know some things were wrong, I refuse to serve anything until everything is as correct as possible". It's probably even right to make super-cautious the default and lesser best-effort-but-more-available be the option.